### ISSUE TITLE
reproduce the paper results?

### ISSUE THREAD

**Issue by yanxurui** [AlexeyAB/darknet#3436]
Hi @AlexeyAB 

Thanks for your amazing work.

![image](https://user-images.githubusercontent.com/12461110/59698274-3ee24580-91e7-11e9-9905-f8d947ef1687.png)

I want to reproduce these results. Sorry if this question has been answered many times.
I have some specific questions:

1. 320 in YOLOv3-320 means width and height in the cfg file. Links to cfg file for YOLOv3-320, YOLOv3-416 and YOLOv3-608 are the same. I assume I only need to modify width and height (default 608) in the cfg by myself accordingly for inference. Right?
2. Are YOLOv3-320, YOLOv3-416 and YOLOv3-608 using the same pretrained weights?
I guess they are trained by setting random=1 such that the input image will be resized to 320 and 608 randomly.
3. The mAP is for the 2014 test-dev set. I can obtain the mAP by uploading the detection results to codalab. However,  I can only find the test-dev 2018 in the participate tab. Are 2018 test-dev and 2014 test-dev the same?
4. How can I measure the FPS?

Thanks.

**Comment by AlexeyAB** [comment]
@yanxurui Hi,

1. Yes, just  change width= height= in cfg-file

2. Yes, for training use pre-trained weights http://pjreddie.com/media/files/darknet53.conv.74 There was trained just one weights file with random=1, and the last 200 iterations was trained for `width=608 height=608 random=0`

3. I don't know. Try to use test-dev 2018

4. just rand detection for 1 image and look at the output measured time in milliseconds. Then divide 1000 ms / measured ms.

**Comment by yanxurui** [comment]
I am still looking for some convincing method to measure FPS because the FPS is an important metric in my report.
Thank you very much.

**Comment by AlexeyAB** [comment]
Try to use command:
`./darknet detector demo ./cfg/coco.data ./cfg/yolov3.cfg ./yolov3.weights test.mp4`

**Comment by yanxurui** [comment]
Hi, @AlexeyAB 

I managed to measure the FPS by detector test.
I also changed the code to include the time of resizing.
```
double time = get_time_point();
        image sized;
        if(letter_box) sized = letterbox_image(im, net.w, net.h);
        else sized = resize_image(im, net.w, net.h);
        layer l = net.layers[net.n - 1];

        //box *boxes = calloc(l.w*l.h*l.n, sizeof(box));
        //float **probs = calloc(l.w*l.h*l.n, sizeof(float*));
        //for(j = 0; j < l.w*l.h*l.n; ++j) probs[j] = (float*)calloc(l.classes, sizeof(float));

        float *X = sized.data;

        //time= what_time_is_it_now();
        network_predict(net, X);
```
I find the resizing takes more than half of the time (20ms) because `resize_image` is too slow.
Why not use `cvResize` from OpenCV?
What interpolation algorithm does `resize_image` use?

Thank you very much.

**Comment by holger-prause** [comment]
Did you tried building with opencv(OPENCV=1 in the makefile)?

**Comment by AlexeyAB** [comment]
@yanxurui 

Yes, there is performance issue for the images - there is an ill-conceived architecture in the original darknet, because Darknet can work without OpenCV.

But in this repository Detection on Video uses OpenCV-resize functions, so there is no bottleneck on CPU. 
GPU-utilization ~90%.

**Comment by yanxurui** [comment]
Hi @holger-prause 
functon `test_detector` doesn't use Opencv's resize even if Opencv is enabled.
